yodeploy/repository.py

```diff
@@ -97,11 +102,29 @@ def get_metadata(self, path):
         with open(meta_fn) as f:
             return json.load(f)
 
-    def put(self, path, fp, metadata=None):
-        '''
-        Store a file (fp).
+    def put(self, path, data, metadata=None):
+        """
+        Store a file (fp), stream, unicode string, or byte string.
```

Comment:
@I still think this docstring is wrong. It doesn't accept file pointers (or handles, this is Python, after all), as far as I can see. It will take file objects, though.